src/domain/community/virtualContributor/vcSettingsPage/VirtualContributorForm.tsx

```diff
@@ -0,0 +1,186 @@
+import { UpdateTagset } from '../../../common/profile/Profile';
+import FormikInputField from '../../../../core/ui/forms/FormikInputField/FormikInputField';
+import { theme } from '../../../../core/ui/themes/default/Theme';
+
+interface VirtualContributorProps {
+  id: string;
+  nameID: string;
+  profile: {
+    id: string;
+    displayName: string;
+    description?: string;
+    tagline: string;
+    tagsets?: Tagset[] | undefined;
+    url: string;
+    avatar?: Visual | undefined;
+  };
+}
+
+interface VirtualContributorFromProps {
+  name: string;
+  nameID: string;
+  description: string;
+  tagline: string;
+  tagsets?: Tagset[];
+  avatar: Visual | undefined;
+}
+
+interface Props {
+  virtualContributor: VirtualContributorProps;
+  avatar: Visual | undefined;
+  onSave?: (virtualContributor: UpdateVirtualContributorInput) => void;
+  title?: string;
+}
+
+export const VirtualContributorForm: FC<Props> = ({
+  virtualContributor: currentVirtualContributor,
+  avatar,
+  onSave,
+}) => {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+
+  const {
+    nameID,
+    profile: { displayName, description, tagline, tagsets },
+  } = currentVirtualContributor;
+
+  const initialValues: VirtualContributorFromProps = {
+    name: displayName,
+    nameID: nameID,
+    description: description ?? '',
+    tagline: tagline,
+    avatar: avatar,
+    tagsets: tagsets,
+  };
+
+  const validationSchema = yup.object().shape({
+    name: nameSegmentSchema.fields?.name || yup.string(),
+    nameID: nameSegmentSchema.fields?.nameID || yup.string(),
+    description: profileSegmentSchema.fields?.description || yup.string(),
+  });
+
+  const getUpdatedTagsets = (updatedTagsets: Tagset[]) => {
+    const result: UpdateTagset[] = [];
+    updatedTagsets.forEach(updatedTagset => {
+      const originalTagset = tagsets?.find(value => value.name === updatedTagset.name);
+      if (originalTagset) result.push({ ...originalTagset, tags: updatedTagset.tags });
+    });
+
+    return result;
+  };
+
+  const [handleSubmit, loading] = useLoadingState(async (values: VirtualContributorFromProps) => {
+    const { tagsets, description, tagline, name, ...otherData } = values;
+    const updatedTagsets = getUpdatedTagsets(tagsets || []);
+
+    const virtualContributor = {
+      ID: currentVirtualContributor.id,
+      profileData: {
+        displayName: name,
+        description,
+        tagline,
+        tagsets: updatedTagsets.map(r => ({
+          ID: r.id,
+          id: undefined,
+          tags: r.tags ?? [],
+        })),
+      },
+      ...otherData,
+    };
+
+    await onSave?.(virtualContributor);
+  });
+
+  const handleBack = () => navigate(-1);
+
+  const backButton = (
+    <Grid item>
+      <WrapperButton variant="default" onClick={handleBack} text={t('buttons.back')} />
+    </Grid>
+  );
+
+  const HostFields = () => (
+    <>
+      <FormikInputField name={'host'} title={'Host'} required readOnly disabled />
```

Comment:
please use double quotes here instead of `{'host'}` ... 
All if this is temporary, right? can you add a comment?